ncdr/importers/grouping.py

```diff
@@ -1,24 +1,30 @@
 import csv
 
-from django.db import transaction
-
 from ..models import DataElement, Grouping
 
 
-@transaction.atomic
-def load_file(file_name, version):
-    with open(file_name, "r", encoding="Windows-1252") as f:
-        f.readline()  # ignore the first line since it's blank
-        rows = list(csv.DictReader(f, delimiter="¬"))
+def load_file(fd, version):
+    fd.readline()  # ignore the first line since it's blank
+    rows = list(csv.DictReader(fd, delimiter="¬"))
+
+    existing_names = {g.name: g.description for g in Grouping.objects.all()}
```

Comment:
It looks like we're only using the name from the existing names. Can we not do

```
set(Grouping.objects.all().values_list('name', flat=True))
```

Without trying it, I'm assuming this is faster and simpler. Note "Without trying it!"

Reply:
Oops, that's some left over from an unnecessary change, good catch!  Fixed and rebased.